### ISSUE TITLE
Odoo 9 - Sale of Component Kit based item does not update Delivered Quantity

### ISSUE THREAD

**Issue by JoeriSoete** [odoo/odoo#10233]
Odoo Version 9.0-20151109

Item with BOM of type "Ship this product as a set of components (kit)"

Create order on Item
Ship item (correctly splits the consumption on its components, possible to assign lots and quantities)
Validate Shipment
Sales order Delivered Quantity stays on 0.
Unable to invoice sales order

**Comment by raycarnes** [comment]
I was unable to replicate this on runbot, selling an "Computer SC234".

After processing the Delivery Order (needed to force availability) the SO showed:

![image](https://cloud.githubusercontent.com/assets/11108754/12060148/0fd97b5a-af1e-11e5-9287-44ca78c7eaff.png)

I then clicked CREATE INVOICE and was able to VALIDATE the resulting Invoice for a single unit, plus automatically added delivery charge.

Perhaps you can try the same - replicate this on runbot in case I misunderstood?

**Comment by ThijsMaenhout** [comment]
I did try this same test and also added a service Item in the Kit.  When a service Item is added the delivered quantity is not updated.

I added the following product in the runbot:

![image](https://cloud.githubusercontent.com/assets/5966582/13284079/3186febc-daf2-11e5-9e6f-d992203ba74f.png)

Can you please verify this?

Kind regards,
Thijs

**Comment by jni-odoo** [comment]
Hello Thijs,

Just want to understand , can I have a service type product in a kit ?

As the service type product does not generate any delivery (as it is a service) , in odoo you will not be able to generate an invoice..

So you have to add a separate line of that service type product in the SO (sale order)..

The other logic can be if I have separate taxes for stockable as well as the service type products, it is better to have the separate line in the SO itself along with your main product line..

Let me know if I am missing something..

Thanks

**Comment by ThijsMaenhout** [comment]
Hi,

A good example where this would apply is:

As a giftshop I would like to sell bottles wrapped up in a gift.  There is an item : GIFT .  this has a phantom BOM wich contains 2 bottles (stockable items), 1 wrap (stockable item), 10 mins of work (service item).    Now I would have the correct costing price on the GIFT.

I only want to show the GIFT on the invoice.

Kind regards,
Thijs

**Comment by jni-odoo** [comment]
As I understand you are selling for example this kit in 100 $...

These 100 $ includes your packaging time (service time) , you are not taking extra money for that..

So you have to maintain a time sheet entry of that 10 mins time , so for example you sell 3 kits and spent 30 mins of time , then in your time tracking you have to maintain how much money that worth for those 30 mins ,rather then selling it..and then accordingly you decides the selling price..

In short you want to maintain cost of your time and track it so you can decide the sell price , rather then selling that time..(that is why you have to user odoo timesheets)

For stock able products you are able to invoice easily.. 

I hope I got your point here ..

Thanks

**Comment by JoeriSoete** [comment]
Hi,

 

We understand why you would only allow stock items … 

But it would be handy if components/service items would be addable … 

For instance, consider in the previous example of Thijs, the need to add a label and lint in the gift item … 

It adds a cost, admittedly a small one, and possibly we would not keep it as a stock item, thus a service cost could be a neat/clean solution.

Plus … time-tracking is not our primary concern … only as a cost element.

 

Secondly, we’ve found an issue concerning different UoM’s in the KIT-BOM …

Consider following setup :

 



 

In the sales order 



 

Shipping the sales order



 

Validating results in following warning (non passable)

 

 

 

(tested on run-bot with a skeleton item, and on our local install with completely configured items)

 

Regards,

 

Joeri SOETE

**Comment by sle-odoo** [comment]
Hi,
The original issue is fixed, even in v9. I created kit1 containing comp1 and comp2, sold it to cust1, once forcing the delivery order the quantity delivered is set to 1.

Thank you for the report,
Regards
